### PR TITLE
Improved Logging for Vulnerability Detector

### LINKED ISSUE #20928: Improve the logging messages of the new vulnerability detector
## Description 

During the E2E tests of the new vulnerability detector module, it was found that some messages aren't clear enough to understand what is the cause of the issue. For example

```
2023/12/19 01:34:05 wazuh-modulesd:vulnerability-scanner[13458] databaseFeedManager.hpp:258 at operator()(): ERROR: Error processing message: Invalid message
```

It would be desirable that the log messages below contain agent info. Debug messages with more details are required to help the developers to fix any problem with the module.

```
2023/12/19 18:47:07 wazuh-modulesd:vulnerability-scanner[29036] packageScanner.hpp:62 at operator()(): DEBUG: SCAN -> Package: openssl, Version: 1.1.1w-0+deb11u1, CVE: CVE-2004-0112, Ver: 0, RLT: 0.9.7d-1, RLTE:
```

```
2023/12/19 18:47:07 wazuh-modulesd:vulnerability-scanner[29036] packageScanner.hpp:139 at operator()(): DEBUG: MATCH Package: openssl, CVE: CVE-2004-0112 by default status.
```

```
2023/12/19 18:47:07 wazuh-modulesd:vulnerability-scanner[29036] resultIndexer.hpp:53 at handleRequest(): DEBUG: Processing key: CVE-2002-0659
```

It would be nice to have information about why the package is not vulnerable. 

```
2023/12/20 16:15:40 wazuh-modulesd:vulnerability-scanner[402] packageScanner.hpp:154 at handleRequest(): DEBUG: Scanning package: vim-common, CNA: redhat
2023/12/20 16:15:40 wazuh-modulesd:vulnerability-scanner[402] packageScanner.hpp:154 at handleRequest(): DEBUG: Scanning package: vim-filesystem, CNA: redhat
2023/12/20 16:15:40 wazuh-modulesd:vulnerability-scanner[402] packageScanner.hpp:154 at handleRequest(): DEBUG: Scanning package: gpm-libs, CNA: redhat
2023/12/20 16:15:40 wazuh-modulesd:vulnerability-scanner[402] packageScanner.hpp:154 at handleRequest(): DEBUG: Scanning package: vim-enhanced, CNA: redhat
```

The report message may be too verbose for INFO level 

```
2023/12/19 16:39:39 wazuh-modulesd:vulnerability-scanner: INFO: REPORT: 1:[001] (Debian) any->vulnerability-scanner:{"vulnerability":{"category":"Packages","classification":"CVSS","cve":"CVE-2002-0659","description":"The ASN1 library in OpenSSL 0.9.6d and earlier, and 0.9.7-beta2 and earlier, allows remote attackers to cause a denial of service via invalid encodings.","enumeration":"CVE","package":{"architecture":"amd64","build_version":"","checksum":"","description":"Secure Sockets Layer toolkit - cryptographic utility","install_scope":"","install_time":" ","license":"","name":"openssl","path":" ","reference":"","size":1471,"type":"deb","version":"1.1.1w-0+deb11u1"},"reference":"http://www.cert.org/advisories/CA-2002-23.html, http://www.kb.cert.org/vuls/id/748355, ftp://ftp.caldera.com/pub/security/OpenLinux/CSSA-2002-033.0.txt, ftp://ftp.caldera.com/pub/security/OpenLinux/CSSA-2002-033.1.txt, ftp://ftp.freebsd.org/pub/FreeBSD/CERT/advisories/FreeBSD-SA-02:33.openssl.asc, http://distro.conectiva.com.br/atualizacoes/?id=a&anuncio=000516, http://rhn.redhat.com/errata/RHSA-2002-160.html, http://rhn.redhat.com/errata/RHSA-2002-161.html, http://rhn.redhat.com/errata/RHSA-2002-164.html, http://www.iss.net/security_center/static/9718.php, http://www.securityfocus.com/bid/5366","report_id":"","scanner":{"vendor":"Wazuh"},"score":{"base":5.0,"environmental":0.0,"temporal":0.0,"version":"2.0"},"severity":"MEDIUM","status":"Active"}}
```

## DoD

- [x] Find in the code all the messages that don't provide enough details and improve them
- [ ] CI Pass

### PR DESCRIPTION
**Pull request by GabrielEValenzuela**
|Related issue|
|---|
|Closes #20928 |

## Objective
This pull request addresses an essential development task aimed at enhancing the logging functionality of the vulnerability detector in the system. The primary focus is on improving the clarity and informativeness of log messages generated during vulnerability detection processes. Specifically, this fix introduces enhancements to log messages to provide additional context and visibility into the detection process. Notably, the addition of the `info` log level is implemented to signify when a match is found in a package, providing valuable insights for monitoring and troubleshooting.

## Description
- The proposed fix introduces enhancements to log messages generated by the vulnerability detector. The improvements aim to provide more context, detail, and clarity in log entries related to the detection of vulnerabilities.

## Quality Assurance
No tests have been added to validate the quality of this implementation

<details><summary>ScanBuild report</summary>

Command:

```bash
scan-build-15 -enable-checker alpha.core.CastSize -enable-checker alpha.core.CastToStruct -enable-checker alpha.core.Conversion -enable-checker alpha.core.IdenticalExpr -enable-checker alpha.core.SizeofPtr -enable-checker alpha.core.TestAfterDivZero --exclude external make TARGET=server DEBUG=1 -j$(nproc)
```

```bash

Done building server

scan-build: No bugs found.

```

</details>

<details><summary>Clang-Format</summary>

```bash
---------------------------------
PASSED: Clang-format check passed
---------------------------------
```
</details>

### Setup
- Clone this branch 
- Compile with `TARGET=server`
- Install manager
- Set `wazuh_modules.debug=2` and restart manager
- Optional: Install indexer
- Connect an agent to the manager
- Wait until the scan ends

<details><summary>Show log</summary>

```text
2023/12/08 09:42:38 wazuh-modulesd:vulnerability-scanner[40979] packageScanner.hpp:62 at operator()(): DEBUG: SCAN -> Package: tar, Version: 1.34+dfsg-1ubuntu0.1.22.04.1, CVE: CVE-2022-48303, Ver: 0, RLT: 1.34+dfsg-1ubuntu0.1.22.10.1, RLTE: 
2023/12/08 09:42:38 wazuh-modulesd:vulnerability-scanner[40979] packageScanner.hpp:107 at operator()(): DEBUG: MATCH Package: tar, Version: 1.34+dfsg-1ubuntu0.1.22.04.1, CVE: CVE-2022-48303, Ver: 0, RLT: 1.34+dfsg-1ubuntu0.1.22.10.1, RLTE:
[...]
2023/12/08 09:43:34 wazuh-modulesd:vulnerability-scanner[40979] packageScanner.hpp:62 at operator()(): DEBUG: SCAN -> Package: dpkg, Version: 1.21.1ubuntu2.2, CVE: CVE-2022-1664, Ver: 0, RLT: 1.21.9ubuntu1, RLTE: 
2023/12/08 09:43:34 wazuh-modulesd:vulnerability-scanner[40979] packageScanner.hpp:107 at operator()(): DEBUG: MATCH Package: dpkg, Version: 1.21.1ubuntu2.2, CVE: CVE-2022-1664, Ver: 0, RLT: 1.21.9ubuntu1, RLTE

//New format
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] resultIndexer.hpp:53 at handleRequest(): DEBUG: Processing and publish key: CVE-2021-3520
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:51 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libidn2' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:166 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libidn2' with CVE Numbering Authorities (CNA) 'redhat' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:69 at operator()(): DEBUG: Scanning package - 'libidn2' (Installed Version: 2.2.0-1.el8, Security Vulnerability: CVE-2019-18224). Identified vulnerability: Version: 0. Required Version Threshold: 2.2.0-1.el8.aarch64. Required Version Threshold (or Equal): N/A.
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:115 at operator()(): DEBUG: Match found, the package 'libidn2', is vulnerable to 'CVE-2019-18224'. Current version: '2.2.0-1.el8' (less than '2.2.0-1.el8.aarch64' or equal to 'N/A').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:190 at handleRequest(): DEBUG: Vulnerability scan for package 'libidn2' on Agent '007' has completed.
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] inventorySync.hpp:139 at handleRequest(): DEBUG: Inserting agent package key: node01_007_fd44bffdb31aef1b8244a6737bb518485a3ed4a3 -> CVE-2019-18224
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] sendReport.hpp:66 at handleRequest(): DEBUG: REPORT: 1:[007] (0635b7aa6181) any->vulnerability-scanner:{"vulnerability":{"category":"Packages","classification":"CVSS","cve":"CVE-2019-18224","description":"idn2_to_ascii_4i in lib/lookup.c in GNU libidn2 before 2.1.1 has a heap-based buffer overflow via a long domain string.","enumeration":"CVE","package":{"architecture":"x86_64","build_version":"","checksum":"","description":"Library to support IDNA2008 internationalized domain names","install_scope":"","installed":"2021-09-15T14:17:30.000Z","license":"","name":"libidn2","path":" ","reference":"","size":287762,"type":"rpm","version":"2.2.0-1.el8"},"reference":"https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=12420, https://github.com/libidn/libidn2/commit/e4d1558aa2c1c04a05066ee8600f37603890ba8c, https://github.com/libidn/libidn2/compare/libidn2-2.1.0...libidn2-2.1.1, http://lists.opensuse.org/opensuse-security-announce/2019-12/msg00008.html, http://lists.opensuse.org/opensuse-security-announce/2019-12/msg00009.html, https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/JDQVQ2XPV5BTZUFINT7AFJSKNNBVURNJ/, https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MINU5RKDFE6TKAFY5DRFN3WSFDS4DYVS/, https://seclists.org/bugtraq/2020/Feb/4, https://security.gentoo.org/glsa/202003-63, https://usn.ubuntu.com/4168-1/, https://www.debian.org/security/2020/dsa-4613","report_id":"","scanner":{"vendor":"Wazuh"},"score":{"base":7.5,"environmental":0.0,"temporal":0.0,"version":"2.0"},"severity":"High","status":"Active"}}
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] resultIndexer.hpp:53 at handleRequest(): DEBUG: Processing and publish key: CVE-2019-18224
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:51 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libkcapi' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:166 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libkcapi' with CVE Numbering Authorities (CNA) 'redhat' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:190 at handleRequest(): DEBUG: Vulnerability scan for package 'libkcapi' on Agent '007' has completed.
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:51 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libpwquality' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:166 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'libpwquality' with CVE Numbering Authorities (CNA) 'redhat' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:48 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:190 at handleRequest(): DEBUG: Vulnerability scan for package 'libpwquality' on Agent '007' has completed.
2024/01/03 10:03:49 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:51 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'elfutils-libelf' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
2024/01/03 10:03:49 wazuh-modulesd:vulnerability-scanner[9058] packageScanner.hpp:166 at handleRequest(): DEBUG: Initiating a vulnerability scan for package 'elfutils-libelf' with CVE Numbering Authorities (CNA) 'redhat' on Agent '0635b7aa6181' (ID: '007', Version: 'v4.7.0').
```

</details>

### Check summary
- [x] Style and quality of SCA.
- [x] Documentation clear.
- [x] Functionality optimizated.

### CODE REVIEW

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/inventorySync.hpp [hunk 1]
```diff
@@ -118,7 +118,7 @@ class InventorySync final : public AbstractHandler<std::shared_ptr<ScanContext>>
                 {
                     insertListString.append(value);
                     logDebug2(WM_VULNSCAN_LOGTAG,
-                              "Updating agent package kv: %s - %s",
+                              "Updating agent package kv: %s -> %s",
```
Comment (Dwordcito) [inline src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/inventorySync.hpp:121]:
```suggestion
                              "Updating agent package key: %s -> %s",
```

Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/inventorySync.hpp]: Thanks! Solved

File: src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp [hunk 2]
```diff
@@ -48,19 +48,28 @@ class PackageScanner final : public AbstractHandler<std::shared_ptr<ScanContext>
...
             for (const auto& version : *callbackData.versions())
             {
                 const std::string packageVersion {data->packageVersion()};
-                std::string versionString {version->version() ? version->version()->str() : ""};
-                std::string versionStringLessThan {version->lessThan() ? version->lessThan()->str() : ""};
+                std::string versionString {version->version() ? version->version()->str() : "N/A"};
+                std::string versionStringLessThan {version->lessThan() ? version->lessThan()->str() : "N/A"};
                 std::string versionStringLessThanOrEqual {version->lessThanOrEqual() ? version->lessThanOrEqual()->str()
-                                                                                     : ""};
+                                                                                     : "N/A"};
```
[the remark this answers is not in the document]
Comment (Dwordcito) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp:67]: image

[the remark this answers is not in the document]
Comment (GabrielEValenzuela) [reply src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/packageScanner.hpp]: Done!

**Review by Dwordcito — APPROVED** [review]

**Pull request merged**